default/python/universe_generation/empires.py

```diff
@@ -95,26 +85,80 @@ def min_planets_in_vicinity_limit(num_systems):
+
+        A good list of home systems are at least the specified minimum number of jumps apart,
+        with the best minimum system merit of all such lists picked randomly from the ''systems_pool''.
+
+        Algorithm:
+        Make several attempts to find systems that match the condition
+        of being at least min_jumps apart.
+        Use the minimum merit of the best num_home_system systems found
+        as the merit of the current best set of systems.
+        On each attempt use the merit of the current best set of home
+        systems to truncate the pool of candidates.
+        """
+
+        # precalculate the system merits
+        for system in systems_pool:
+            if system not in self.system_merit:
+                self.system_merit[system] = calculate_home_system_merit(system)
+
+        # In the list of systems sorted by merit, if the first num_home_systems are all min_jumps
+        # apart, then the merit of the num_home_systems nth system is the
+        # best_case_merit_lower_bound that can be achieved.
+        best_case_merit_lower_bound = sorted(self.system_merit.values())[self.num_home_systems - 1]
+        current_merit_lower_bound = 0
+
+        best_candidate = []
+
+        # cap the number of attempts to the smaller of the number of systems in the pool, or 100
+        attempts = min(100, len(systems_pool))
+
+        # Copy systems_pool to avoid aliasing of the caller's systems_pool variable
+        local_pool = set(systems_pool)
+
+        while attempts:
+            candidate = []
+            # use a local pool of all candidate systems better than the worst threshold merit
+            local_pool = {s for s in local_pool if self.system_merit[s] > current_merit_lower_bound}
+            if len(local_pool) < self.num_home_systems:
+                break
+
+            while local_pool:
+                member = random.choice(list(local_pool))
+                candidate.append(member)
+
+                # remove all neighbors from the local pool
+                local_pool -= set(get_systems_within_jumps(member, min_jumps))
+
+            if len(candidate) < self.num_home_systems:
+                break
+
+            # Calculate the merit of the current attempt.  If it is the best so far
+            # keep it and update the merit_threshold
+            merit_system = sorted([(self.system_merit[s], s)
+                                   for s in candidate])[:self.num_home_systems]
+            merit = merit_system[-1]
+            if merit > current_merit_lower_bound:
+                current_merit_lower_bound = merit
+                best_candidate = [s for (_, s) in merit_system]
+
+            # quit if it isn't possible to improve the current accepted list
+            if merit >= best_case_merit_lower_bound:
+                break
+
+            attempts = min(attempts - 1, len(local_pool))
```

Comment:
Isn't ```len(local_pool)``` equivalent to zero here? Otherwise, we wouldn't have left the while loop above?